File: python/grade-school/grade_school.py
from collections import defaultdict

class School(object):
    def __init__(self):
        self.children = defaultdict(list)

    def add_student(self, name, grade):
        self.children[grade].append(name)

    def roster(self):
         return [child for grade in sorted(self.children) for child in sorted(self.children[grade])]

    def grade(self, grade_number):
        return sorted(self.children[grade_number])

File: kotlin/grade-school/src/main/kotlin/GradeSchool.kt
class School {

    private var students = mutableMapOf<Int, List<String>>()

    fun add(student: String, grade: Int) {
        val gradeStudents = students.getOrPut(grade) { emptyList() }
        students.put(grade, (gradeStudents + student).sorted())
    }

    fun grade(grade: Int) = students[grade]?: emptyList()

    fun roster() = students.keys
        .sorted()
        .map { grade -> students[grade]!! }
        .flatMap{ it }
}
